simulation.cpp

```diff
@@ -373,7 +373,9 @@ void state_manager::process_commands() {
 				vehicle->MoverParameters->DistCounter = 0.0;
 				vehicle->MoverParameters->WheelFlat = 0.0;
 				vehicle->MoverParameters->AlarmChainFlag = false;
-				vehicle->MoverParameters->OffsetTrackH = 0.0;
+				if (vehicle->MyTrack->iCategoryFlag & 1) { // reset horizontal offset only for rail
+					vehicle->MoverParameters->OffsetTrackH = 0.0;
```

Comment:
Co to zmienia?

Reply:
Pojazdy samochodowe (czyli na drogach, kategoria 2) mają ustawiane przesunięcie poziome względem osi toru (offsetTrackH), żeby po ruszały się prawym pasem drogi w kierunku jazdy. Wyzerowanie im tego parametru po wykolejeniu powoduje ustawienie samochodu w osi drogi, a tego byśmy nie chcieli.